packages/dara-components/js/smart/chat/chat.tsx

```diff
@@ -80,14 +124,29 @@ function Chat(props: ChatProps): JSX.Element {
     const [showChat, setShowChat] = React.useState(false);
     const theme = useTheme();
 
+    const onUpdate = (newValue: Message[]): void => {
+        // If the new value is longer than the old value, we can assume that a new message was added
+        if (newValue.length > value.length) {
+            const newMessage = newValue[newValue.length - 1];
+            const users = getAllUsersInChat(newValue);
```

Comment:
Is it better to have a copy of all the users in a state in the component or calculate it whenever a message is sent?

Reply:
I think that's premature optimization, realistically how often will this be invoked? You're not really gaining enough to warrant the increased complexity in code. 
Let's start with how it is right now, if we notice it causing issues and measure that this is a bottleneck then we can revisit.